src/server/scripts/EasternKingdoms/Karazhan/karazhan.cpp

```diff
@@ -218,28 +218,24 @@ class npc_barnes : public CreatureScript
             switch (m_uiEventId)
             {
                 case EVENT_OZ:
-                    if (OzDialogue[count].textid)
-                        text = OzDialogue[count].textid;
+                    text = OzDialogue[count].textid;
                     if (OzDialogue[count].timer)
                         TalkTimer = OzDialogue[count].timer;
                     break;
 
                 case EVENT_HOOD:
-                    if (HoodDialogue[count].textid)
-                        text = HoodDialogue[count].textid;
+                    text = HoodDialogue[count].textid;
                     if (HoodDialogue[count].timer)
                         TalkTimer = HoodDialogue[count].timer;
                     break;
 
                 case EVENT_RAJ:
-                    if (RAJDialogue[count].textid)
-                        text = RAJDialogue[count].textid;
+                    text = RAJDialogue[count].textid;
                     if (RAJDialogue[count].timer)
                         TalkTimer = RAJDialogue[count].timer;
                     break;
             }
 
-            if (text)
                 CreatureAI::Talk(text);
```

Comment:
I checked the script, and this is always called with valid parameters `count [0, 3]` and valid the struct data
```cpp
static Dialogue OzDialogue[] =
static Dialogue HoodDialogue[] =
static Dialogue RAJDialogue[] =
```
so we can get rid of the `ifs` here. Funnily the only thing needed to make this work was removing the `if (text)` check, but it can be cleaned up a little more

```suggestion
            switch (m_uiEventId)
            {
                case EVENT_OZ:
                    text = OzDialogue[count].textid;
                    TalkTimer = OzDialogue[count].timer;
                    break;
                case EVENT_HOOD:
                    text = HoodDialogue[count].textid;
                    TalkTimer = HoodDialogue[count].timer;
                    break;
                case EVENT_RAJ:
                    text = RAJDialogue[count].textid;
                    TalkTimer = RAJDialogue[count].timer;
                    break;
                default:
                    return;
            }

            CreatureAI::Talk(text);
```